src/multitaper.jl

```diff
@@ -0,0 +1,583 @@
+spec2 = mt_spectrogram(signal2, mt_config, 250) # same `mt_config`, different signal, different overlap
+```
+"""
+function mt_spectrogram(signal::AbstractVector, mt_config::MTConfig{T},
+                        n_overlap::Int=mt_config.n_samples >> 1) where {T}
+    config = MTSpectrogramConfig{T}(length(signal), mt_config, n_overlap)
+    X = allocate_output(config)
+    return mt_spectrogram!(X, signal, config)
+end
+
+
+#####
+##### Multitapered cross spectral matrix
+#####
+
+"""
+    CrossSpectral{T,F}
+
+Fields:
+
+* `values::T`: `n_channels` x `n_channels` x `length(freq)` array
+* `freq::F`: frequencies; accessed by `freq(::CrossSpectral)`
+
+"""
+struct CrossSpectral{T,F}
+    values::T
+    freq::F
+end
+
+freq(c::CrossSpectral) = c.freq
+
+struct MTCrossSpectraConfig{T,T1,T2,T3,T4,F,T5,T6,C<:MTConfig{T}}
+    n_channels::Int
+    weighted_evals::T1
+    x_mt::T2
+    demean::Bool
+    mean_per_channel::T3
+    demeaned_signal::T4
+    freq::F
+    freq_range::T5
+    freq_inds::T6
+    ensure_aligned::Bool
+    mt_config::C
+end
+
+"""
+    MTCrossSpectraConfig{T}(n_channels, n_samples; fs=1, demean=true, low_bias=true, freq_range=nothing, nw=4, kwargs...) where T
+
+Creates a configuration object used for [`mt_cross_spectral!`](@ref) as well as [`mt_coherence!`](@ref).
+
+
+* `n_samples`: the number of samples to be used as input when computing multitaper periodograms
+    with this configuration. Used for pre-allocating temporary buffers.
+* `fs`: the number of samples per second of the input signal
+* `demean`: if `true`, the channelwise mean will be subtracted from the input signals before the cross spectral values are computed.
+* `low_bias`: if `true`, keeps only tapers with eigenvalues > `0.9`.
+* `freq_range`: if `nothing`, all frequencies are retained. Otherwise, only frequencies between `first(freq_range)` and `last(freq_range)` are retained.
+* `ensure_aligned = T == Float32 || T == Complex{Float32}`: perform an extra copy to ensure that the FFT output is memory-aligned
+
+Any keywords accepted by [`MTConfig`](@ref) may be passed here.
+"""
+function MTCrossSpectraConfig{T}(n_channels, n_samples; fs=1, demean=true, low_bias=true,
+                                 freq_range=nothing, nw=4, ntapers=2 * nw - 1,
+                                 ensure_aligned = T == Float32 || T == Complex{Float32},
+                                 kwargs...) where {T}
+    if demean
+        mean_per_channel = Vector{T}(undef, n_channels)
+        demeaned_signal = Matrix{T}(undef, n_channels, n_samples)
+    else
+        mean_per_channel = nothing
+        demeaned_signal = nothing
+    end
+
+    window = dpss(n_samples, nw, ntapers)
+    evals = dpsseig(window, nw)
+
+    if low_bias
+        keep_evals_mask = evals .> 0.9
+        window, evals = window[:, keep_evals_mask], evals[keep_evals_mask]
+    end
+    ntapers = size(window, 2)
+
+    denom = fs * sum(evals) / 2
+    weighted_evals = evals ./ denom
```

Comment:
Could you provide a reference for weighting each Slepian taper before using it to estimate the spectra? I'm unfamiliar with this, and it's not used in any other spectral estimation code that I've seen.

Reply:
I based this off of MNE's `csd_multitaper` method. I saw MATLAB's `pmtm` also provides an option to weight by the eigenvalues, https://www.mathworks.com/help/signal/ref/pmtm.html?searchHighlight=psd%20multitaper&s_tid=srchtitle#btultt2-method, but I don't know specifically how they do it (and that is not for cross-spectra). I found it discussed briefly in Section 3.3.1 of this reference as well:

Candy, J. V. “MULTITAPER SPECTRAL ESTIMATION: An Alternative to the Welch Periodogram Approach.” Lawrence Livermore National Lab. (LLNL), Livermore, CA (United States), August 19, 2019. https://doi.org/10.2172/1560107.

Would it be better to provide an option about whether or not to weight it, and how to do so?

Reply:
Did you figure out if this should be `nfft` or the size of the signal? It seems to me like it should be `nfft`.

Reply:
I am pretty sure it should be `nfft`; I've filed an issue with MNE and someone there agreed as well. So I've changed it here to be `nfft`.